### PR TITLE
Pass workflow_name and input_folder to DbResultWriter

### PR DESCRIPTION
**Pull request by wolny**
Parsing logs in `DbResultWriter` was error prone. This PR fixes this issue by passing the necessary context directly via the constructor or `BaseJob.run()` method

### CODE REVIEW

File: batchlib/workflows/mean_and_sum_cell_analysis.py [hunk 1]
```diff
@@ -34,8 +34,8 @@ def mean_and_sum_cell_analysis(config):
     merger(config.folder, config.folder, force_recompute=True)
 
     # 4.) run all the usual workflow summaries
-    identifiers = modify_identifiers(identifiers)
-    stat_names = [idf + stat_name.replace('serum_', '')
-                  for idf in identifiers for stat_name in DEFAULT_PLOT_NAMES]
-    name = "MeanAndSumCellAnalysisWorkflow"
-    workflow_summaries(name, config, t0, stat_names)
+    # identifiers = modify_identifiers(identifiers)
```
Comment (wolny) [inline batchlib/workflows/mean_and_sum_cell_analysis.py:37]: @constantinpape I just need to figure out what to do with this. `workflow_summaries()` will be correctly invoked as part of `run_cell_analysis` for mean and sum, so why do we need to call it explicitly here as well?

Comment (constantinpape) [reply batchlib/workflows/mean_and_sum_cell_analysis.py:37]: No for `mean_and_sum` `feature_identifers is not None`, so `workflow_summaries` will not be invoked.

Comment (wolny) [reply batchlib/workflows/mean_and_sum_cell_analysis.py:37]: ok, I got it now. It's the merged table that also needs to be written to the DB. Ok I'll try to make it work on the DB job level

Comment (constantinpape) [reply batchlib/workflows/mean_and_sum_cell_analysis.py:37]: exactly

Comment (wolny) [reply batchlib/workflows/mean_and_sum_cell_analysis.py]: :white_check_mark:

**Comment by wolny** [comment]
This is good to go BTW, I've tested it

**Comment by constantinpape** [comment]
Thanks!

**Pull request merged**